packages/back-end/test/data-generator/new-generator.ts

```diff
@@ -378,7 +379,6 @@ function purchase(
     ),
   });
   trackExperiment(data, res, "confirmation-email", sim);
-  sim[parseInt(data.userId)] += normalInt(res.value - 10, res.value + 10);
```

Comment:
This was raised by the type checker. `sim` is an object holding simulation params. Not sure what this line is/was supposed to achieve..

Reply:
Thanks for finding this, looks like a bug that I landed. The intent is to make it so that the `"confirmation-email"` experiment changes the likelihood that user returns, which is handled by the `userRetention` object.

The line that you've deleted does nothing, I would think, but we should find a way to restore this functionality. I think the following might be the right way to do this:

```
sim.dataTables.userRetention[parseInt(data.userId)] += normalInt(res.value - 10, res.value + 10);
```

See similar functionality here: https://github.com/growthbook/growthbook/pull/2273/files#diff-a7e9d8cb1781870b7a4cb2d0dd0b12ef8cb0eac22867638f9d037a15898605c1L490